### ISSUE TITLE
Enabling Gyro seems to disable other gamepad inputs

### ISSUE THREAD

**Issue by EliteMasterEric** [SagaPDev/Godot-SDL-Gyro#11]
On windows, in my project, it seems inconsistent, but either Gyro will work or Godot button/axis inputs will work, not both together

**Comment by SagaPDev** [comment]
curious, what controller are you using?

**Comment by EliteMasterEric** [comment]
Playstation 4

**Comment by ThePhantomMask** [comment]
It happens to me too, having the PS4 controller

**Comment by SagaPDev** [comment]
I've been meaning to debug this with a friend's dualshock 4, gyro is managed by SDL and buttons should be handled independently by Godot own input system this happens both over bluetooth and wired right?

**Comment by ThePhantomMask** [comment]
Yes, but Godot's Input System is also based on SDL

**Comment by SagaPDev** [comment]
For what that understand Godot only uses the SDL mapping database, input itself is handled separately (dinput or xinput not sure) there is no actual SDL code in Godot.
https://x.com/reduzio/status/1770037294486958413

**Comment by ThePhantomMask** [comment]
Yes, use that

**Comment by ThePhantomMask** [comment]
What causes the problem?

**Comment by SagaPDev** [comment]
My theory, is that when the dualshock 4 is accessed from SDL and another input system at the same time it causes some kind of conflict ( haven't been able to get a hold on a ps4 controller to test)

**Comment by ThePhantomMask** [comment]
Which is strange because they used to work together, if there is a bug in Godot's input system that conflicts?

**Comment by SagaPDev** [comment]
> Which is strange because they used to work together, if there is a bug in Godot's input system that conflicts?

so this issue is recent? the plugin used to fine for you at some point?

**Comment by ThePhantomMask** [comment]
Yes

**Comment by SagaPDev** [comment]
> Yes

Sorry for having more question that answers, but do you have any clue of what could have changed? this project hasn't have mayor updates since i posted it, did you update your Godot, does it work again in previous versions? (minimum for this plugin should be 4.1)
any  info on this could help me pinpoint the issue

**Comment by ThePhantomMask** [comment]
Well...I used Godot 4.2 (it worked before)

**Comment by ThePhantomMask** [comment]
Well...I used Godot 4.2 (it worked before)

**Comment by ThePhantomMask** [comment]
Maybe add C++ bindings for the Godot script API for Godot 4.2

**Comment by alvina-dr** [comment]
Hello @SagaPDev, same problem here, absolutely unable to figure out a solution to make the inputs work at the same time as the gyro, I don't know if you've had any clue since then ? It would help a lot, we're making a student project and are kinda stuck on this (also can't seem to make it work with 4.2 and older versions of the gdextension). Honestly I'd be okay rolling the project back to an older version, but I can't find a setup that works, maybe @ThePhantomMask has info about exactly what worked for them ?)

**Comment by SagaPDev** [comment]
@alvina-dr unfortunately i haven't been able to reproduce the issue still, I'm using 4.2 and the add-on should be compatible from then onward, I've been asking other users with dual-shock for for clues to the issue but for some it works fine

for clarification this seem to be happening on windows and with dualshock 4, right?

**Comment by alvina-dr** [comment]
@SagaPDev I'm on windows and using **joycons** from the switch ! 

I've just discovered that part of the input were blocked by Steam being open (apparently there's some conflict with it ?). I had reached a point where even in a new clean projet the joycon inputs weren't working anymore... It's at least something, it might help someone, who knows.

And once that's out of the way, I can reference new controls in input actions using input detection (in the project with your gdextension in it), which I couldn't before. But then when I play it, the events from the joycon aren't printed. I'm printing using : 
```
func _unhandled_input(event):
	print(event)
```

After that, when I go again to try and use input detection to reference new controls, it doesn't work, just like before in my first comment 🤔

I tried testing a few other things and found this : 
Using PS4 dualshock, everything works well, I get both input and gyroscope (sadly I **_have_** to use joycons, we're making a project that needs to be accessible and joycons are a must have for us).
Using Xbox controller I only get inputs (I don't even know if there's a gyroscope, so it might be normal, but it was worth a try aha).

**Comment by ThePhantomMask** [comment]
It still doesn't work for me

**Comment by SagaPDev** [comment]
well thanks @alvina-dr, every bit of info helps, but now i'm even more confused as to why it is not working for some people, even while using the same device, i can't see a pattern.

Steam spawns a virtual game-pad to handle their input remapping suite, this often takes place as the 1st controller, so it might cause issues

If your actions in Godot Input Map are mapped to "all devices" it shouldn't cause problems as it will respond to any device connected, but if you're making a multiplayer game you would want to specify, it just a matter of being aware of what devices are the OS detecting

So, know that you mention Joycons, do they work for you in a clean project? i have a Switch-pro, but windows input system does not read them correctly (random inputs), Godot relies on the OS input system and on windows they're broken

not sure if this affects joy-cons as well

There is a talk about it here godotengine/godot#47874

**Comment by alvina-dr** [comment]
@SagaPDev it's not exactly that, I don't have chaotic inputs, I just don't get any at all. Anyway, we have to complete the project by friday morning, so we decided to switch to Unity, it's a bit sad but we'll have to try again with godot on another project :/ 

Thanks for you help though !

**Comment by ThePhantomMask** [comment]
> > It still doesn't work for me
> 
> could you send me and example project were the issue is present so i can take a look at it?

As soon as I have time, I will send you a simple project with the add-on

**Comment by mcerdeira** [comment]
Same issue here. I'm using Godot 4.2.1 on Windows 11 with Switch Pro Controllers.

**Comment by SagaPDev** [comment]
i haven't been able to recreate this specific issue but i found something similar
running a project that handles input via de input singleton in the process function:
```
func _process(_delta):
	if Input.is_action_just_pressed("jump"):
		***
```
will fail if a steam a input layout is in is active but empty
*this happes automatically when a steam game is minimized
*this would happen only with my pro-controller a dualshock 3 i have seems unaffected"

in projects using _unhandled_input function
```
func _unhandled_input(event):
	if event.is_action_pressed("jump"):
		***
```

the input will be read no matter the status of steam

**Comment by meloncolle** [comment]
I know this hasn't been updated in a while, but I wanted to share my findings struggling with this same issue.

Using Godot 4.4.1. I have 2 gyro-capable controllers, PS5 & an 8BitDo SN30pro+. When connected via bluetooth, I had the issue of button/axis inputs not working for both controllers once gyro was enabled. This persisted even when opening other Godot projects (but inputs were working in other applications). I had to disconnect and reconnect the controller in order to restore button/axis inputs in Godot.

When I tried connecting the 8BitDo via USB, I was able to get both button/axes and gyro inputs at the same time, but the button inputs were really glitchy ([this seems to be an unrelated issue though](https://github.com/godotengine/godot/issues/47874)).

Finally, connecting the PS5 controller via USB resulted in proper button/axes and gyro inputs at the same time. It's nice to have found a workaround, but I feel like since the required setup is so specific, it's not really viable to ship my project with motion controls. I appreciate all your work looking into this issue so far, it would be great if a solution could be found yet.

**Comment by SagaPDev** [comment]
Thanks for the report, i wanted but couldn't get back to this project, but in the mean time someone has been working on adding SDL3 support to Godot https://github.com/godotengine/godot-proposals/issues/2829#issuecomment-3172541863 so well probably see this implemented into the next releases of Godot soon
